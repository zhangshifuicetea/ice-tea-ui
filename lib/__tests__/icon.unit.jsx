import React from 'react';
import * as renderer from 'react-test-renderer';
import Icon from '../icon/icon';
import {mount} from 'enzyme';

describe('icon', () => {
    it('render', () => {
        const json = renderer.create(<Icon name={'wechat'}/>).toJSON();
        expect(json).toMatchSnapshot();
    });

    it('onClick', () => {
        const fn = jest.fn();
        const component = mount(<Icon name={'wechat'} onClick={fn}/>)
        component.find('span').simulate('click');
        expect(fn).toBeCalled();
    })
});
