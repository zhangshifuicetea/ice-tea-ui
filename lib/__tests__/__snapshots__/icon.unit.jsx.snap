// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render 1`] = `
<span
  className="ice-tea-icon-wrapper"
>
  <svg
    className="ice-tea-icon ice-tea-icon-wechat"
    style={
      Object {
        "fill": undefined,
      }
    }
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;

exports[`icon snapshot 1`] = `
<span
  className="ice-tea-icon-wrapper"
>
  <svg
    className="ice-tea-icon ice-tea-icon-wechat"
    style={
      Object {
        "fill": undefined,
      }
    }
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;
